device/iot-device-client/src/main/java/com/microsoft/azure/sdk/iot/device/auth/IotHubSasToken.java

```diff
@@ -168,7 +168,7 @@ else if(this.signature != null && this.expiryTime != 0L && this.scope!= null)
         }
         else
         {
-            return null;
```

Comment:
This should never be hit, so instead of returning null which would cause NPEs everywhere, throw an exception instead

Reply:
Given that this is a public method, could this introduce a breaking change if a customer is using this method?

Reply:
I'd chalk this class up as not public considering we never ask a user to instantiate it directly. At most, people may use this as a utility class of sorts, and I don't expect them to hit this code path even in that case